emissionsapi/openapi.yml

```
@@ -89,6 +118,17 @@ components:
         type: integer
         minimum: 0
       example: '0'
+    interval:
+      in: query
+      name: interval
+      description: Specify the interval length in which to aggregate data.
+        Specify the time interval length for which data is being aggregated as
+        accepted by [PostgreSQL's date_trunc
+        ](https://www.postgresql.org/docs/9.1/functions-datetime.html#FUNCTIONS-DATETIME-TRUNC)
+        function like `day` or `week`.
+      schema:
+        type: string
```

You could go for a [enum](https://swagger.io/docs/specification/data-models/enums/) here.

```suggestion
        type: string
        enum: [day, week, month, quarter, year]
```
This would have multiple advantages: 

* You do not have to reference to PostgreSQL.
* You can limit the enum to reasonable values.
* You can use connexion as a first validator of the values, see 

```bash
$ curl -X GET "http://127.0.0.1:5000/api/v1/statistics.json?interval=banana" -H "accept: application/json"
{
  "detail": "'banana' is not one of ['day', 'week', 'month', 'quarter', 'year']\n\nFailed validating 'enum' in schema:\n    {'enum': ['day', 'week', 'month', 'quarter', 'year'], 'type': 'string'}\n\nOn instance:\n    'banana'",
  "status": 400,
  "title": "Bad Request",
  "type": "about:blank"
}
```

---

The downside is that we would need to keep this list twice. But in any case, it shouldn't return a 500 when the input is invalid. I'll likely use the enum, although with different values than suggested.